DataFormats/SiStripCluster/interface/SiStripApproximateClusterCollection.h

```diff
@@ -90,7 +90,7 @@ class SiStripApproximateClusterCollection {
   void reserve(std::size_t dets, std::size_t clusters);
   Filler beginDet(unsigned int detId);
 
-  const_iterator begin() const { return const_iterator(this); }
+  const_iterator begin() const { return const_iterator(clusters_.empty() ? nullptr : this); }
```

Comment:
To me
```suggestion
  const_iterator begin() const { return clusters_.empty() ? end() : const_iterator(this); }
```
would be clearer as then the reader would not need to check the behavior of the non-default `const_iterator` constructor to understand that in this special case `begin() == end()`.